File: AdminPanel1/AdminPanel1.Web/Modules/ProductImages/ProductImages/ProductImagesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel1.ProductImages.Columns;

[ColumnsScript("ProductImages.ProductImages")]
[BasedOnRow(typeof(ProductImagesRow), CheckNames = true)]
public class ProductImagesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ProductName { get; set; }
    [EditLink]
    public string Image { get; set; }
}

File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/ProductImages/ProductImagesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductImagesRow } from "./ProductImagesRow";

export interface ProductImagesColumns {
    Id: Column<ProductImagesRow>;
    ProductName: Column<ProductImagesRow>;
    Image: Column<ProductImagesRow>;
}

export class ProductImagesColumns extends ColumnsBase<ProductImagesRow> {
    static readonly columnsKey = 'ProductImages.ProductImages';
    static readonly Fields = fieldsProxy<ProductImagesColumns>();
}